tests/test_enterprise/management/test_backfill_ecu_table_user_foreign_key.py

```diff
@@ -0,0 +1,129 @@
+"""
+Tests for the Django management command `backfill_ecu_table_user_foreign_key`.
+"""
+
+from unittest.mock import patch
+
+import ddt
+import factory
+from pytest import mark
+
+from django.contrib import auth
+from django.core.management import call_command
+from django.db import DatabaseError
+from django.db.models import signals
+from django.test import TestCase
+
+from enterprise.models import EnterpriseCustomer, EnterpriseCustomerUser
+from test_utils import factories
+
+EXCEPTION = "DUMMY_TRACE_BACK"
+
+User = auth.get_user_model()
+
+
+@mark.django_db
+@ddt.ddt
+class CreateEnterpriseCourseEnrollmentCommandTests(TestCase):
+    """
+    Test command `backfill_ecu_table_user_foreign_key`.
+    """
+    command = 'backfill_ecu_table_user_foreign_key'
+
+    @factory.django.mute_signals(signals.post_save)
+    def setUp(self):
+        super().setUp()
+        self.cleanup_test_objects()
+
+        for i in range(12):
+            factories.UserFactory(username=f'user-{i}')
+
+        self.customer = factories.EnterpriseCustomerFactory(
+            name='alpha',
+        )
+
+        users = User.objects.all()
+
+        # Make a bunch of users for an ENT customer
+        for _index, user in enumerate(users[0:12]):
+            factories.EnterpriseCustomerUserFactory(
+                user_id=user.id,
+                enterprise_customer=self.customer,
+            )
+
+        self.addCleanup(self.cleanup_test_objects)
+
+    def cleanup_test_objects(self):
+        """
+        Helper to delete all instances of role assignments, ECUs, Enterprise customers, and Users.
+        """
+        EnterpriseCustomerUser.objects.all().delete()
+        EnterpriseCustomer.objects.all().delete()
+        User.objects.all().delete()
+
+    def test_copies_user_id_to_user_fk(self):
+        ecu = EnterpriseCustomerUser.objects.first()
+        ecu.user_fk = None
+
+        # use bulk_update to prevent save() method from setting user_fk to user_id
+        EnterpriseCustomerUser.objects.all().bulk_update([ecu], ['user_fk'])
+        ecu.refresh_from_db()
+        assert ecu.user_fk is None
+        call_command(self.command)
+        assert EnterpriseCustomerUser.objects.first().user_fk is EnterpriseCustomerUser.objects.first().user_id
```

Comment:
nit: another way to do this is to explicitly refresh the `ecu` model instance from the DB, and then assert that the user FK == the user id:
```python
ecu.refresh_from_db()
assert ecu.user_fk == ecu.user_id
```

Reply:
done